### DISCUSSION TITLE
X16 charset patch alternative... or not

### DISCUSSION THREAD

**Discussion by kelli217** [pzembrod/cc64#24] [Ideas]
I see that there are some settings that have to be applied in order to accommodate the characters that C uses which aren't present in PETSCII. I also see that one needs these settings even in the X16 version. However, the X16 does have an ISO mode, specifically ISO-8859-15 if memory serves. The lower 127 codes are identical to plain ASCII. That would seem to remove the need for the charset patch on that platform.

However, it would also seem to require a complete overhaul of every string function in the compiler, so I can certainly see why you wouldn't want to use ISO mode.

**Comment by pzembrod** [comment]
The discussion came up in the context of Stefan's X16 Edit:
https://www.commanderx16.com/forum/index.php?/topic/390-text-editor/page/4/&tab=comments#comment-6537

I expect to eventually be able to create an ISO flavour of cc64 on my roadmap. It would have to be that, as far as I can see - a second flavour, as in, a separte binary. What's preventing me at the moment is that the Forth target compiler I use for the X16 VolksForth runs on the C64 and thus produces PETSCII. There are ASCII VolksForth platforms where the target compiler can run - CP/M, MSDOS, AtariST - and unifying the different VolksForth platforms including the target compiler is what we are working towards. An ISO-compatible X16 VolksForth should become easier from some point, and that would unlock an ISO cc64. The primary sources of both are, after all, already ASCII.
